frontend/src/lib/forms/UserTypeahead.svelte

```diff
@@ -12,29 +13,58 @@
   // eslint-disable-next-line @typescript-eslint/no-redundant-type-constituents
   export let value: string;
   export let debounceMs = 200;
+  export let isAdmin: boolean = true;
 
   let input = writable('');
   $: $input = value;
+  let typeaheadResults = deriveAsync(
+    input,
+    isAdmin ? _userTypeaheadSearch : _orgMemberTypeaheadSearch,
+    [],
+    debounceMs);
 
-  let typeaheadResults = deriveAsync(input, _userTypeaheadSearch, [], debounceMs);
+  const dispatch = createEventDispatcher<{
+      selectedUserId: string | null;
+  }>();
 
   function formatResult(user: SingleUserTypeaheadResult): string {
     const extra = user.username && user.email ? ` (${user.username}, ${user.email})`
                 : user.username ? ` (${user.username})`
                 : user.email ? ` (${user.email})`
+                : user.id ? ` (${user.id})`
                 : '';
     return `${user.name}${extra}`;
   }
 
+  function getInputValue(user: SingleUserTypeaheadResult | SingleUserInMyOrgTypeaheadResult): string {
+    if ('email' in user && user.email) return user.email;
+    if ('username' in user && user.username) return user.username;
+    if ('name' in user && user.name) return user.name;
+    return '';
+  }
+
 </script>
 
 <FormField {id} {label} {error} {autofocus} >
   <div use:overlay={{ closeClickSelector: '.menu li'}}>
-    <PlainInput style="w-full" {id} bind:value type="text" autocomplete="off" />
+    <PlainInput
+      style="w-full"
+      bind:value {id}
+      type="text"
+      autocomplete="off"
+      keydownHandler={() => {dispatch('selectedUserId', null)}}
```

Comment:
This is going to cause a LOT of events, most of which aren't necessary. I'd rather use a Svelte store subscription to do the dispatching here, because Svelte stores automatically ignore it when similar values are inserted. Rather than try to explain what I mean, I'll just push a commit to implement what I'm suggesting.